pinot-clients/pinot-jdbc-client/src/main/java/org/apache/pinot/client/PinotPreparedStatement.java

```diff
@@ -51,7 +51,7 @@ public PinotPreparedStatement(PinotConnection connection, String query) {
     if (!DriverUtils.queryContainsLimitStatement(_query)) {
       _query += " " + LIMIT_STATEMENT + " " + _maxRows;
     }
-    _query = DriverUtils.enableNullHandling(_connection, _query);
+    _query = _connection.enableQueryOptions(_query);
```

Comment:
I am a bit confused on how would the user control whether to use pinot-v1 or pinot-v2.
enable null handling is configured by option but are we planning to do so as well for v1 vs. v2?

should we create a different connection string format for v1 or v2, for example:
- for V1: `jdbc:pinot:default_database`
- for V2: `jdbc:pinot_v2:default_database`

Reply:
V1 vs V2 is controlled by an option as well as the cluster config. QueryOptionKey.USE_MULTISTAGE_ENGINE is an existing key.

I did consider whether this is useful long term because I'm not sure if the intent is to support both versions in the future.